source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+Now this code can be run with:
+
+.. code-block:: bash
+
+   singularity exec --nv nv-tf-22.07.sif ./fashion.py
+
+Or run it interactively in the container line-by-line with:
+
+.. code-block:: bash
+
+   singularity shell --nv nv-tf-22.07.sif 
+
+If there is an output in the terminal running the python code similar to:
+
+.. code-block:: bash
+
+   2022-07-29 11:53:24.017428: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1532] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 30987 MB memory:  -> device: 0, name: Tesla V100-PCIE-32GB, pci bus id: 0000:00:06.0, compute capability: 7.0
+
+this means the GPU is being used for your computations.
+
+Also, by wrapping the singularity command in a shell script called ``fashion.sh`` and adding the appropriate ``#SBATCH`` commands at the top, the script can be submitted to the batch system with ``sbatch fashion.sh``.
+
+
+.. _jupyter-notebooks:
+
+Running jupyter notebooks
+=========================
+
+Many users prefer working in interactive notebooks during development of their models. Here an example is shown of running tensorflow in a jupyter notebook. There is also a more general section on jupyter notebooks :ref:`here <jupyter-notebook-section>`.
+
+.. tip::
+   Make sure you use the GPU version and not the CPU version of your software in the container.
+
+We start with the image from the previous subsection, the tensorflow container from the Nvidia repository with the added examples: ``nv-tf-22.07.sif``. This image also contains jupyter by default.
+
+.. code-block:: bash
+
+    singularity shell --nv nv-tf-22.07.sif
+
+Before starting the notebook, we have open a tunnel to forward the port on which the python kernel communicates to the local machine where the user works. In this way, the notebook can be openened in the browser:
+
+.. code-block:: bash
+
+   ssh -NL 8888:wn-gp-01:8888 USERNAME@spider.surfsara.nl
+
+where USERNAME is your username and ``wn-gp-01`` should changed to the node on which the python kernel is running. This tunneling command has to be running in a seperate terminal, and ensures the communication from port 8888 (right hand side) on the remote machine is forwarded to port 8888 (left hand side) on the local machine. The port that is given when you start the jupyter notebook defaults to 8888, but if it is already in use, the value will be different.
+
+Once the tunnel is open, start the notebook in a new terminal with:
+
+.. code-block:: bash
+
+   ssh USERNAME@spider.surfsara.nl
+   srun --partition=gpu_v100 -gpus v100:1 --time=12:00:00 --x11 --pty bash -i -l
```

Comment:
A small typo in gpus parameter: ``srun --partition=gpu_v100 --gpus v100:1 --time=12:00:00 --x11 --pty bash -i -l``